### PR TITLE
Additional Device Informations for ble

### PR DESCRIPTION
**Pull request by elral**
Added some methods for setting more device informations.
See also Issue #335

### CODE REVIEW

File: libraries/Bluefruit52Lib/src/services/BLEDis.h [hunk 1]
```diff
@@ -46,24 +46,56 @@ class BLEDis : public BLEService
...
+        uint8_t _pnp_id_length;
+      };
+      const uint8_t _strarr_length[9];
     };
 
   public:
     BLEDis(void);
 
     void setModel(const char* model);
+    void setModel(const char* model,uint8_t length);
```
Comment (hathach) [inline libraries/Bluefruit52Lib/src/services/BLEDis.h]: for using this API with length, I would suggest to 
1. change the type of 1st parameter to `const uint8_t*` e.g `setModel(const uint8_t model, uint8_t length)`
2. also please implement the `const char*` as inline function as well.

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp [hunk 2]
```diff
@@ -51,51 +51,164 @@ BLEDis::BLEDis(void)
...
     {
-      BLECharacteristic chars(UUID16_CHR_MODEL_NUMBER_STRING+i);
+      BLECharacteristic chars(UUID16_CHR_SYSTEM_ID+i);
       chars.setTempMemory();
-
       chars.setProperties(CHR_PROPS_READ);
-      chars.setFixedLen(strlen(_strarr[i]));
-
+      if (_strarr_length[i]) {
+        chars.setFixedLen(_strarr_length[i]);
```
Comment (hathach) [inline libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: 3. should only use the _strarr_legnth[i], there is no point to use the strlen() function anymore.

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp [hunk 3]
```diff
@@ -51,51 +51,164 @@ BLEDis::BLEDis(void)
...
+      } else {
+        chars.setFixedLen(strlen(_strarr[i]));
+      }
       VERIFY_STATUS( chars.begin() );
-      chars.write(_strarr[i]);
+      if (_strarr_length[i]) {
```
Comment (hathach) [inline libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: same as above

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp [hunk 4]
```diff
@@ -51,51 +51,164 @@ BLEDis::BLEDis(void)
...
+        chars.write(_strarr[i], _strarr_length[i]);
+      } else {
+        chars.write(_strarr[i]);
+      }
+      
     }
   }
 
+  if ( _strarr[arrcount(_strarr)-1] != NULL )
```
Comment (hathach) [inline libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: 4. I am not familliar with PNP_ID, but why this characteristic isn't included in the above loop

File: libraries/Bluefruit52Lib/src/services/BLEDis.h (hunk 1 above)
Comment (elral) [reply libraries/Bluefruit52Lib/src/services/BLEDis.h]: 1. Yes, you are right. Then inside function it must be "_model = (const char*)model;"
2. Sorry, I don't know what to change, or do you mean "_model = (const char*)model;"

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp (hunk 2 above)
Comment (elral) [reply libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: Yes, you are rigth.

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp (hunk 3 above)
Comment (elral) [reply libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: Yes, you are rigth.

File: libraries/Bluefruit52Lib/src/services/BLEDis.cpp (hunk 4 above)
Comment (elral) [reply libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: Last characteristic from the loop is 0x2A2A (IEEE_REGULATORY_CERTIFICATION), PNP_ID is 0x2A50. So there is a big gap between last entry from loop and PNP_ID. I had no better idea than to do it seperatly.

Comment (hathach) [reply libraries/Bluefruit52Lib/src/services/BLEDis.cpp]: ah i see the reason now.

**Pull request merged**